packages/flutter_tools/test/general.shard/intellij/intellij_validator_test.dart

```diff
@@ -373,6 +374,29 @@ void main() {
     expect(validator.pluginsPath, '/path/to/JetBrainsToolboxApp.plugins');
   });
 
+  testWithoutContext('IntelliJValidatorOnMac.installed() handles FileSystemExceptions)', () async {
+    const FileSystemException exception = FileSystemException('cannot list');
+    final FileSystem fileSystem = _ThrowingFileSystem(exception);
+
+    final FakeProcessManager processManager = FakeProcessManager.empty();
+
+    final Iterable<DoctorValidator> validators = IntelliJValidatorOnMac.installed(
+      fileSystem: fileSystem,
+      fileSystemUtils: FileSystemUtils(fileSystem: fileSystem, platform: macPlatform),
+      userMessages: UserMessages(),
+      plistParser: FakePlistParser(<String, String>{
+        'JetBrainsToolboxApp': '/path/to/JetBrainsToolboxApp',
+        'CFBundleIdentifier': 'com.jetbrains.toolbox.linkapp',
+      }),
+      processManager: processManager,
+    );
+
+    expect(validators.length, 1);
+    final DoctorValidator validator = validators.first;
+    expect(validator, isA<ValidatorWithResult>());
+    expect(validator.title, 'Cannot determine if IntelliJ is installed');
```

Comment:
uber nit: Consider also testing against the validator message. This would make it impossible for the test to pass if we got a validator from some other source that happens to use the same title.